test/common/test_context.go

```diff
@@ -260,7 +268,29 @@ func (tcb *TestContextBuilder) Build() *TestContext {
 	return testContext
 }
 
-func newSMServer(smEnv env.Environment, wg *sync.WaitGroup, fs []func(ctx context.Context, smb *sm.ServiceManagerBuilder, env env.Environment) error) (*testSMServer, storage.Repository) {
+func NewSMListener() (net.Listener, error) {
+	minPort := 8100
+	maxPort := 9999
+
+	rand.Seed(time.Now().UnixNano())
+	port := rand.Intn(maxPort-minPort) + minPort
+
+	retries := 10
+
+	var listener net.Listener
+	var err error
+	for ; retries >= 0; retries-- {
```

Comment:
why the retries

Reply:
In case the random selected random port is already selected, I think it would be best to try a different port, so that the execution does not stop. But I see now that I haven't placed the logic for random port number generation inside the loop, will fix it now..